activities/models/hashtag.py

```diff
@@ -0,0 +1,145 @@
+import re
+from datetime import date, timedelta
+from typing import Dict, List
+
+import urlman
+from asgiref.sync import sync_to_async
+from django.db import models
+from django.utils import timezone
+
+from stator.models import State, StateField, StateGraph, StatorModel
+
+
+class HashtagStates(StateGraph):
+    outdated = State(try_interval=300)
+    updated = State(externally_progressed=True)
+
+    outdated.transitions_to(updated)
+
+    @classmethod
+    async def handle_outdated(cls, instance: "Hashtag"):
+        """
+        Computes the stats and other things for a Hashtag
+        """
+        from .post import Post
+
+        posts_query = Post.objects.local_public().tagged_with(instance)
+        total = await posts_query.acount()
+
+        today = timezone.now().date()
+        # TODO: single query
+        total_today = await posts_query.filter(
+            created__gte=today,
+            created__lte=today + timedelta(days=1),
+        ).acount()
+        total_month = await posts_query.filter(
+            created__year=today.year,
+            created__month=today.month,
+        ).acount()
+        total_year = await posts_query.filter(
+            created__year=today.year,
+        ).acount()
+        if total:
+            if not instance.stats:
+                instance.stats = {}
+            instance.stats.update(
+                {
+                    "total": total,
+                    today.isoformat(): total_today,
+                    today.strftime("%Y-%m"): total_month,
+                    today.strftime("%Y"): total_year,
+                }
+            )
+            print(f"Hashtag {instance.hashtag} stats={instance.stats}")
+            instance.stats_updated = timezone.now()
+            await sync_to_async(instance.save)()
+        else:
+            print(f"Hashtag {instance.hashtag} - No Totals")
+        return cls.updated
+
+
+class Hashtag(StatorModel):
+
+    # Normalized hashtag without the '#'
+    hashtag = models.SlugField(primary_key=True, max_length=100)
+
+    # Friendly display override
+    name_override = models.CharField(max_length=100, null=True, blank=True)
+
+    # Should this be shown in the public UI?
+    public = models.BooleanField(default=True)
```

Comment:
Is the intent for this to be the moderation of hashtags into, say, trending? If so, we may want to have this be three way (nullable BooleanField) with the default of null meaning "unreviewed, don't show"

Reply:
I went with binary state to treat moderation as a reactive correction with the ability to preempt spammy hashtags by prepopulating. I can change to trinary if you'd prefer.

Reply:
Now trinary with a `Config.system` to specify if unreviewed is treated as public True or False